snuba/datasets/errors_replacer.py

```diff
@@ -242,6 +244,97 @@ def _build_event_tombstone_replacement(
     )
 
 
+def _build_event_merge_replacement(
+    message: Mapping[str, Any],
```

Comment:
You are using the message only to receive the new_group_id, transaction_id and the project_id. I'd consider replacing message with these three arguments even if it increases annoyingly the number of arguments. 
Given the size of this module, having precise arguments with their type defined would make it harder to mess up on the call sites.